### PR TITLE
text dataset pipeline 

### PR DESCRIPTION
**Pull request by sahilgupta2105**
Completes the missing pieces for integrating the text dataset into the project pipeline, adds a new test dataset called "spam.csv", which contains binary data for spam/ ham messages, the size of dataset is kept small to around 1K rows to ensure that running the test is not time-consuming,

picked the dataset from here: https://www.kaggle.com/uciml/sms-spam-collection-dataset 

## Primary type of change:
- [ x] New feature.
- [ x] Bug fix.
- [ x] Refactor.
- [ ] Documentation.
- [ ] Build/ devops.
- [ x] Test.
- [ ] Other.

## Checklist:
- [ x] I have actually ran this code locally to make sure it works.
- [ x] I have pulled and merged the latest `main` into my feature branch.
- [ x] I have ran the existing [tests](https://github.com/aiqc/aiqc/new/main/.github#how-to-run-tests), and inspected areas that my code touches.
- [ x] I have updated the tests to include my changes.
- [ ] I have included updates to the documentation.
- [ ] I have built the documentation files `make html`.

> Describe the env you tested on: mac ios 13, python 3.7.10

## Contains breaking changes:
- [ ] Yes (explain what functionality it breaks).
- [ x] No

### CODE REVIEW

File: aiqc/__init__.py [hunk 1]
```diff
@@ -597,6 +597,22 @@ def sorted_file_list(dir_path:str):
...
+		if (dataset.dataset_type == 'image'):
+			raise ValueError("\n Dataset class does not support get_main_file() method for `image` data type,\n")
+
+		file = File.select().join(Dataset).where(
+			Dataset.id==id, File.file_type=='tabular', File.file_index==0
+		)[0]
+		return file
+
+
+	def get_main_tabular(id:int):
```
Comment (aiqc) [inline aiqc/__init__.py:612]: Nice! I hated the old approach!

File: aiqc/__init__.py [hunk 2]
```diff
@@ -1018,14 +1040,18 @@ def to_numpy(
...
 			word_counts, feature_names = Dataset.Text.get_feature_matrix(df)
-			return word_counts, feature_names
+			return word_counts.todense()
 
 
 		def get_feature_matrix(
 			dataframe:object
 		):
-			count_vect = CountVectorizer()
+			count_vect = CountVectorizer(max_features = 200)
```
Comment (aiqc) [inline aiqc/__init__.py:1054]: Maybe we could expose a `**kwargs` for situations like this

File: aiqc/__init__.py (hunk 1 above)
Comment (aiqc) [inline aiqc/__init__.py:612]: Tested that this works with both Dataset.Text and Dataset.Tabular

File: aiqc/tests/__init__.py [hunk 3]
```diff
@@ -1125,6 +1198,8 @@ def make_test_queue(name:str, repeat_count:int=1, fold_count:int=None):
 		queue = make_test_queue_keras_multiclass(repeat_count, fold_count)
 	elif (name == 'keras_binary'):
 		queue = make_test_queue_keras_binary(repeat_count, fold_count)
+	elif (name == 'keras_text_binary'):
```
Comment (aiqc) [inline aiqc/tests/__init__.py:1201]: I'll add a line for this to the PR template tests

**Review by aiqc — APPROVED** [review]
Love the dataset! Congrats on punching through!
spam

File: aiqc/__init__.py (hunk 2 above)
Comment (sahilgupta2105) [reply aiqc/__init__.py:1054]: sure thing, will create an issue for this

File: aiqc/tests/__init__.py (hunk 3 above)
Comment (sahilgupta2105) [reply aiqc/tests/__init__.py:1201]: thanks

**Pull request merged**